keyless/circuit-data/templates/helpers/jwt_field_parsing.circom

```diff
@@ -128,3 +128,59 @@ template ParseJWTFieldWithUnquotedValue(maxKVPairLen, maxNameLen, maxValueLen) {
     }
 }
 
+template ParseEmailVerifiedField(maxKVPairLen, maxNameLen, maxValueLen) {
+    signal input field[maxKVPairLen]; // ASCII
+    signal input name[maxNameLen];
+    signal input value[maxValueLen];
+    signal input field_len; // ASCII
+    signal input name_len;
+    signal input value_index; // index of value within `field`
+    signal input value_len;
+    signal input colon_index; // index of colon within `field`
+
+
+    ParseJWTFieldSharedLogic(maxKVPairLen, maxNameLen, maxValueLen)(field, name, value, field_len, name_len, value_index, value_len, colon_index);
+
+
+    signal char_before_value <== SelectArrayValue(maxKVPairLen)(field, value_index-1);
+    signal before_is_quote      <== IsEqual()([char_before_value, 34]);
+    signal before_is_whitespace <== isWhitespace()(char_before_value);
+    signal before_is_whitespace_or_quote<== OR()(before_is_quote, before_is_whitespace);
+    // Check the char before is either quote or whitespace, OR that it is the colon
+    (1 - before_is_whitespace_or_quote)*(value_index-1-colon_index) === 0;
```

Comment:
Can this be done with logical operators instead?

Reply:
Looks like AND(NOT(A), B) where A = before_is_whitespace_or_quote, and B = value_index-1-colon_index

Reply:
The comment is correct. The constraint checks that one of the following is true:
* `before_is_whitespace_or_quote` is 1
* `value_index - 1` is equal to `colon_index`, i.e., it is the colon.

I'm not sure how to succinctly express this with logical operators. It would have to be something like `OR(before_is_whitespace_or_quote, IsEqual(value_index-1, colon_index)`; since circom doesn't allow nested template instantiations, it would be pretty wordy to expand this out. But you're free to make a PR if you have a better idea

Reply:
Oh I see. I think you're expressing OR(A,B) here using NOT(AND(A,B)) via De Morgan's law. 

Similar to my comment above, the code expresses AND(NOT(A), ~B), where ~B = NOT(B) where B is value_index-1 == colon_index. This is equivalent to NOT(AND(A,B)) = OR(A,B). So I think we're both describing the same thing